go/fn/object_test.go

```diff
@@ -10,6 +10,78 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+type GVKTest struct {
+	group   string
+	version string
+	kind    string
+	resultA bool
+	resultB bool
+	resultC bool
+}
+
+func TestGVK(t *testing.T) {
+	input := []byte(`
+apiVersion: apps/v3
+kind: StatefulSet
+metadata:
+  name: my-config
+spec:
+  volumeClaimTemplates:
+    - metadata:
+        labels:
+          testkey: testvalue
+`)
+
+	input_noGroup := []byte(`
+apiVersion: v3
+kind: StatefulSet
+metadata:
+  name: my-config
+spec:
+  volumeClaimTemplates:
+    - metadata:
+        labels:
+          testkey: testvalue
+`)
+
+	input_differentKind := []byte(`
+apiVersion: apps/v1
+kind: Service
+metadata:
+  name: my-config
+spec:
+  volumeClaimTemplates:
+    - metadata:
+        labels:
+          testkey: testvalue
+`)
+	testcases := map[string]GVKTest{}
+	testcases["1"] = GVKTest{"apps", "", "StatefulSet", true, true, false}
+	testcases["2"] = GVKTest{"", "v3", "StatefulSet", true, true, false}
+	testcases["3"] = GVKTest{"apps", "v3", "StatefulSet", true, true, false}
+	testcases["4"] = GVKTest{"", "", "StatefulSet", true, true, false}
+	testcases["5"] = GVKTest{"", "", "", true, true, true}
+	testcases["6"] = GVKTest{"appWrong", "", "", false, true, false}
+	testcases["7"] = GVKTest{"", "", "Service", false, false, true}
+	testcases["8"] = GVKTest{"", "v1", "", false, false, true}
```

Comment:
For readability, it may be better to (a) have the map string be more descriptive of what we are testing and (b) create this struct inline:

```
	testCases := map[string]struct {
		group   string
		version string
		kind    string
		resultA bool
		resultB bool
		resultC bool
	}{
		"description of which case we are trying to test here": {
			group:   "apps",
			version: "",
			kind:    "StatefulSet",
			resultA: true,
			resultB: true,
			resultC: true,
		},
		"description of second case we are trying to test here": {
			group:   "",
			version: "v3",
			kind:    "StatefulSet",
			resultA: true,
			resultB: true,
			resultC: true,
		},
	}
```

The descriptions will help the reviewer and future code maintainers understand what is being tested.